### DISCUSSION TITLE
Shared positional shadow atlases

### DISCUSSION THREAD

**Discussion by markusneg** [godotengine/godot-proposals#7089] [3D]
Currently, positional shadow maps are completely per-Viewport. Thus, with multiple Viewports, shadows of the same Light3D's are calculated repeatedly, leading to a significant drop of frame-rate in scenes with some positional lights. Technically, it shouldn't be too difficult to share shadow atlases between Viewports. A caveat could be different cull layers for GeometryInstances but in most cases the lights should see the same geometries.

**Comment by AThousandShips** [comment]
This would require that all atlas settings match as well, I don't know if the checking and mapping would be worth it, and on top of that reference countin I would suspect since shared

**Comment by Calinou** [comment]
The camera may be located at very different positions when using multiple viewports (e.g. for split screen), so using the same atlas quadrants isn't always viable.

**Comment by m4rr5** [comment]
They might indeed be at very different positions, but there is also a use case where they are not. For example when games use multiple monitors that are not positioned in a flat plane. This is a very common setup for racing games, with a triple screen setup where side monitors are typically angled at 45-60 degrees. An option to share those atlases (and tweak them to cover the three viewports, possibly with a bit of bias towards the one in the middle) would certainly be a welcome optimization.

**Reply by Calinou** [reply]
> This is a very common setup for racing games, with a triple screen setup where side monitors are typically angled at 45-60 degrees.

This should be achievable with less overhead using a single viewport that spans 3 monitors and a distortion shader (or [alternative camera projection](https://github.com/godotengine/godot-proposals/issues/3779)).
